app/support_centre_handlers.py

```diff
@@ -0,0 +1,130 @@
+
+        self.log_entry(request, display_region + '/find-a-support-centre')
+
+        if display_region == 'cy':
+            locale = 'cy'
+            # TODO Add welsh translation
+            page_title = 'Find a support centre'
+        else:
+            locale = 'en'
+            page_title = 'Find a support centre'
+        return {
+            'display_region': display_region,
+            'page_title': page_title,
+            'locale': locale,
+            'page_url': View.gen_page_url(request)
+        }
+
+    @aiohttp_jinja2.template('support_centre_enter_postcode.html')
+    async def post(self, request):
+        self.setup_request(request)
+        display_region = request.match_info['display_region']
+
+        if display_region == 'cy':
+            locale = 'cy'
+        else:
+            locale = 'en'
+
+        self.log_entry(request, display_region + '/find-a-support-centre')
+
+        data = await request.post()
+
+        try:
+            postcode = ProcessPostcode.validate_postcode(data['form-enter-address-postcode'], locale)
+            logger.info('valid postcode', client_ip=request['client_ip'])
+
+        except (InvalidDataError, InvalidDataErrorWelsh) as exc:
+            logger.info('invalid postcode', client_ip=request['client_ip'])
+            flash_message = FlashMessage.generate_flash_message(str(exc), 'ERROR', 'POSTCODE_ENTER_ERROR', 'postcode')
+            flash(request, flash_message)
+            raise HTTPFound(
+                request.app.router['SupportCentreEnterPostcode:get'].url_for(
+                    display_region=display_region,
+                ))
+
+        raise HTTPFound(
+            request.app.router['SupportCentreListCentres:get'].url_for(
+                display_region=display_region,
+                postcode=postcode
+            ))
+
+
+@support_centre_routes.view(r'/' + View.valid_ew_display_regions + '/find-a-support-centre/{postcode}/')
+class SupportCentreListCentres(View):
+    @aiohttp_jinja2.template('support_centre_list_of_centres.html')
+    async def get(self, request):
+        self.setup_request(request)
+        display_region = request.match_info['display_region']
+
+        if display_region == 'cy':
+            locale = 'cy'
+        else:
+            locale = 'en'
+
+        try:
+            postcode_value = ProcessPostcode.validate_postcode(request.match_info['postcode'], locale)
+            logger.info('valid postcode', client_ip=request['client_ip'])
+
+        except (InvalidDataError, InvalidDataErrorWelsh):
+            logger.info('invalid postcode', client_ip=request['client_ip'])
+            attributes = {
+                'page_title': 'Error',
+                'display_region': display_region,
+                'locale': locale,
+                'page_url': View.gen_page_url(request)
+            }
+            return aiohttp_jinja2.render_template('404.html', request, attributes, status=404)
```

Comment:
I decided to return the 404 because of the url structure - the url is /en/find-a-support-centre/postcode/, so the postcode is a part of the url rather than being in a query string param (and in theory could be accessed without using the form). In effect, each potential postcode has its own url, so a 404 seemed more appropriate.

Reply:
Yes, is a path parameter not a query string parameter. Is debatable as to whether a Bad Request 400 or Not Found 404 as basically we don't have the resource requested by this URL. Was trying to differentiate between a badly formed postcode which is the cause of this 404 and valid postcode returning Not Found in the AD lookup later (see below). 
The client is directed here by the previous POST where they enter their postcode and it is checked and not accepted if not valid but as you say they could in theory just amend the GET URL for this which is why I pointed this out at all.
See what testing say.